vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -37,9 +38,15 @@ class Graph(VizroBaseModel):
     """
 
     type: Literal["graph"] = "graph"
-    figure: CapturedCallable = Field(..., import_path=px)
+    figure: CapturedCallable = Field(
+        ..., import_path=px
+    )  # TODO: needs to be refactored so plotly-independent or extendable - No, as within the boundaries of Graph model
     actions: List[Action] = []
 
+    # Component properties for actions and interactions
+    _input_property: str = PrivateAttr("clickData")
```

Comment:
There's a big correlation between _input_property and _action_validator_factory.

Just to remind that:

_action_validator_factory  argument ("value" in this scenario) represents a default property when the model object is the trigger of the actions chain.
_input_property  argument ("value" in this scenario) represents a default property when the model object is the state of some action.

We definitely need to refine this implementation especially considering that there are cases where `_input_property` doesn't make sense as a single value. Explained in another PR [here](https://github.com/mckinsey/vizro/pull/119/files#r1365118530).

Reply:
Good question, I thought about this as well, and I think the answer is mostly, but not always. You could argue that there can be different triggers to inputs if I am not mistaken, but @petar-qb may have a clearer view on this!

Reply:
So as I said:
> `_action_validator_factory argument` represents a default property when the model object is the `trigger` of the actions chain.
And `_input_property` represents a default property when the model object is the `State` of some action.

Also, we can say that `_output_property` represents a default property when the model object is the `Output` of some action.

The problem with the `_input_property` and `_output_property` approach is that default property `state` and `output` sometimes is not a single value like [here](https://github.com/mckinsey/vizro/pull/119/files#r1365118530) and it also could potentially depend on the concrete predefined action implementation.

The `"not a single value"` (in "_input_property" / "_output_property") problem could be solved by moving dependency of State/Output creation from `_callback_mapping_utils` to additional model methods. 
So for example, instead of having the following code inside the `_get_action_callback_outputs`:

```
    return {
        target: Output(
            component_id=target,
            component_property=model_manager[target]._output_property,  # type: ignore[attr-defined]
            allow_duplicate=True,
        )
        for target in targets
```

We can define `_get_model_action_callback_output` method (and _get_model_action_callback_input) inside the model in the following way:

```
# IMPORTANT: Inside this method we can return multiple outputs if we need.
def _get_model_action_callback_output(self):
    return Output(
            component_id=self.id,
            component_property="value"
            allow_duplicate=True,
        )
``` 

and call it inside the `_callback_mapping_utils` function `_get_action_callback_outputs` in a following way:

```
    return {
        target: target._get_model_action_callback_output()
        for target in targets
```

This still doesn't solve the problem if the inputs/outputs depend on the concrete predefined action implementation which could happen potentially. 

--- 

So what do you think, what's the correct approach?

Reply:
Thanks for the explanation @petar-qb, it makes sense 👍 I don't have an immediate answer to this - will need to spend some time familiarising myself with the actions code again and thinking about the new possibilities that tables bring. And then definitely let's discuss it.

Reply:
Scheduled a meeting to discuss this

Reply:
Outcome of discussion is that:
- we keep things as they are in this PR
- abstraction logic should lie with functions, as we do not want a multitude of models
- refactor later, as this is not public API